sdk/identity/azure-identity/tests/test_cli_credential.py

```diff
@@ -92,6 +92,48 @@ def test_get_token():
     assert token.expires_on == expected_expires_on
 
 
+def test_expires_on_used():
+    """Test that 'expires_on' is preferred over 'expiresOn'."""
+    expires_on = 1602015811
+    successful_output = json.dumps(
+        {
+            "expiresOn": datetime.fromtimestamp(1555555555).strftime("%Y-%m-%d %H:%M:%S.%f"),
+            "expires_on": expires_on,
+            "accessToken": "access token",
+            "subscription": "some-guid",
+            "tenant": "some-guid",
+            "tokenType": "Bearer",
+        }
+    )
+
+    with mock.patch("shutil.which", return_value="az"):
+        with mock.patch(CHECK_OUTPUT, mock.Mock(return_value=successful_output)):
+            token = AzureCliCredential().get_token("scope")
+
+    assert token.expires_on == expires_on
+
+
+def test_expires_on_string():
+    """Test that 'expires_on' still works if it's a string."""
+    expires_on = 1602015811
+    successful_output = json.dumps(
+        {
+            "expires_on": f"{expires_on}",
```

Comment:
do you need the `f` character there? What is that used for?

Reply:
This is the format for a Python [f-string](https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals), and I'm using it here to test the unexpected case where expires_on is read in as a string. Generally shouldn't happen since expires_on is expected to be an integer, but still nice to know that things will still work if it is somehow a string.

Reply:
TIL, thank you!